src/coprocessor/dag/expr/builtin_op.rs

```diff
@@ -0,0 +1,365 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{FnCall, Result, StatementContext};
+use coprocessor::codec::Datum;
+
+impl FnCall {
+    pub fn logic_and(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
+        let arg0 = try!(self.children[0].eval_int(ctx, row));
+        if arg0.is_some() && arg0.unwrap() == 0 {
+            return Ok(Some(0));
+        }
+        let arg1 = try!(self.children[1].eval_int(ctx, row));
+        if arg1.is_some() && arg1.unwrap() == 0 {
+            return Ok(Some(0));
+        }
+        if arg0.is_none() || arg1.is_none() {
+            return Ok(None)
+        }
+        Ok(Some(1))
+    }
+
+    pub fn logic_or(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
```

Comment:
somehow strange naming, `logical or`.

Never mind if this is for consistency with TiDB code.